gensim/models/word2vec.py

```diff
@@ -131,23 +131,17 @@
 from collections import defaultdict, namedtuple
 from types import GeneratorType
 import threading
-import itertools
+import itertools as it
```

Comment:
-1 on this.

Pros: it makes things slightly more brief.
Cons: it's not a universally accepted abbreviation or convention, so it can make the code more difficult to read.

The pros don't outweigh the cons in my opinion.

Reply:
I agree. See the original PR for a discussion with @gojomo – he prefers `it` because it's more idiomatic.

Reply:
I increasingly see `it` as a common alias for `itertools`, and as the use of the individual functions becomes more prevalent/natural/assumed, having to either – (1) add/remove each function by name in imports as they move in/out of use; or (2) prepend `itertools` everywhere– both add cognitive load & repetitive typing compared to just adopting `import itertools as it`.

Similarly, a lot of our older code imports each function used from `numpy` individually - which given the absolute dominance of the `import numpy as np` convention throughout our domain, just added maintenance overhead & extra-indirection in reading code. I've been gradually using `np.` instead, but the transition is not complete.

Reply:
> both add cognitive load & repetitive typing compared to just adopting import itertools as it

Yes, it's additional cognitive load, but only for one person - the writer of the code, and only once - when the code is written. The alias adds cognitive load for many people - whoever needs to read the code, and for many times over - because code is read far more often than it is written.

Reply:
I believe it's harder to read without consistent alias use, as well. 

In the case where the functions are imported into the module scope (as happens in gensim code of a certain vintage, especially `from itertools import chain` but there have been others), the function name may not be unambiguously from `itertools` - so a reader may have to search for the definition or import to understand, and adding/removing usages is onerous. 

The case where they're always preceded by the full `itertools.` prefix is somewhat better, but sometimes that causes common combined to more-often wrap lines, requiring the reader to parse indention-conventions. Consistently using `it.` everywhere only requires the reader to realize one thing – "this codebase uses it=itertools convention" – and from that moment on, code is both compact & readable.

Reply:
I'm also in favour of explicit `itertools.xyz`, not a fan of 2-letter acronyms. Especially ones that coincide with common English words.

How strongly do you feel about this @gojomo ?

Reply:
@gojomo ping. Unless this is vital for you I'll revert to spelling out `itertools` in full, I concur with @mpenkov.

Reply:
It's not vital to me either way. I find `it` a helpful convention, I've seen it in a growing number of places, and I think that `itertools` is central-enough to idiomatic Python programming that I expect this conventions' prevalence to grow over time, eventually reaching the same status as `np`. But it's ok to wait/follow here rather than lead!

Reply:
OK. Renamed to `itertools` for now, in da8847a0.